### PR TITLE
Move all docs into the docs/ folder.

### PR DESCRIPTION
**Pull request by jverkoey**
This removes the `howto/` folder entirely and makes our github doc layout match our website doc layout more closely.

### CODE REVIEW

File: docs/docsite-index.md [hunk 1]
```diff
@@ -141,13 +141,13 @@ An easy way to create beautiful apps with modular and customizable UI&nbsp;compo
 
     <ul class="icon-list">
       <li class="icon-list-item icon-list-item--guide">
-        <a href="../howto">Read the Development Guide</a>
+        <a href="../docs">Read the Development Guide</a>
```
Comment (shyndman) [inline docs/docsite-index.md]: Should be `href="README.md"`

File: docs/docsite-index.md [hunk 2]
```diff
@@ -141,13 +141,13 @@ An easy way to create beautiful apps with modular and customizable UI&nbsp;compo
...
       </li>
       <li class="icon-list-item icon-list-item--components">
         <a href="../components">View the Component Documentation</a>
       </li>
       <li class="icon-list-item icon-list-item--code">
-        <a href="../howto/tutorial/#sample-code">Explore our Code Samples</a>
+        <a href="../docs/tutorial/#sample-code">Explore our Code Samples</a>
```
Comment (shyndman) [inline docs/docsite-index.md]: You can lose `../docs/`

File: docs/docsite-index.md (hunk 1 above)
Comment (jverkoey) [reply docs/docsite-index.md]: Done.

File: docs/docsite-index.md (hunk 2 above)
Comment (jverkoey) [reply docs/docsite-index.md]: Done.

**Review by yarneo — APPROVED** [review]

**Pull request merged**